### PR TITLE
Add pre-commit with initial codespell hook

### PR DESCRIPTION
**Pull request by rkm**
## Proposed Changes

Adds a pre-commit config with codespell. Contains fixes for all current spelling mistakes.

## Types of changes

What types of changes does your code introduce? Tick all that apply.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation-Only Update (if none of the other choices apply)
  - In this case, ensure that the message of the head commit from the source branch is prefixed with `[skip ci]`

## Checklist

By opening this PR, I confirm that I have:

- [x] Reviewed the [contributing](https://github.com/SMI/SmiServices/blob/master/CONTRIBUTING.md) guidelines for this repository
- [x] Ensured that the PR branch is in sync with the target branch (i.e. it is automatically merge-able)
- [ ] Updated any relevant API documentation
- [ ] Created or updated any tests if relevant
- [x] Created a [news](https://github.com/SMI/SmiServices/blob/master/news/README.md) file
    -   NOTE: This ***must*** include any changes to any of the following files: default.yaml, any of the RabbitMQ server configurations, GlobalOptions.cs
- [x] Listed myself in the [CONTRIBUTORS](https://github.com/SMI/SmiServices/blob/master/CONTRIBUTORS.md) file 🚀
- [x] Requested a review by one of the repository maintainers

## Issues
-

### CODE REVIEW

**Comment by rkm** [comment]
I'd like to add some installation/usage instructions for pre-commit. I can either do that in this PR or in a separate one?

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  - repo: https://github.com/codespell-project/codespell
+    rev: "v2.0.0"
+    hooks:
+      - id: codespell
+        args:
+          [
+            --ignore-words, .meta/codespell-ignores,
+            --exclude-file, .meta/codespell-excludes,
+            # NOTE(rkm 2021-07-13) Ignore stanford english.all.3class.distsim.crf.ser.gz file
+            --ignore-regex, 'ser\.gz',
```
Comment (jas88) [inline .pre-commit-config.yaml:13]: Shouldn't this regex have a tail anchor, '\.ser\.gz$'?

**Comment by jas88** [comment]
> I'd like to add some installation/usage instructions for pre-commit. I can either do that in this PR or in a separate one?

Sounds good - in this PR? Looks good to me otherwise.

**Comment by tznind** [comment]
I'm happy if you are @jas88 go ahead and approve it.

**Review by jas88 — APPROVED** [review]

**Review by jas88 — APPROVED** [review]

**Pull request merged**